src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -34,7 +34,7 @@ public ForegroundNotificationService()
             _workQueue = new PriorityQueue();
             _lastProcessedTimeInMS = Environment.TickCount;
 
-            Debug.Assert(IsValid());
+            Contract.ThrowIfFalse(IsValid());
```

Comment:
I'm pretty sure we saw this assert during a scenario involving shared projects because CPS set a `JoinableTaskFactory` `SynchronizationContext`.  @davkean do you remember?